tests/python_client/testcases/test_collection.py

```diff
@@ -2161,6 +2162,254 @@ def test_load_partitions_release_collection(self):
+        expected: raise exceptions
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        # load with non-number replicas
+        error = {ct.err_code: 0, ct.err_msg: f"but expected one of: int, long"}
+        collection_w.load(replica_number=get_non_number_replicas, check_task=CheckTasks.err_res, check_items=error)
+
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("replicas", [-1, 0, None])
+    def test_load_replica_invalid_number(self, replicas):
+        """
+        target: test load partition with invalid replica number
+        method: load with invalid replica number
+        expected: raise exception
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        collection_w.load(replica_number=replicas)
+        replicas = collection_w.get_replicas()[0]
+        groups = replicas.groups
+        assert len(groups) == 1
+        assert len(groups[0].shards) == 2
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_load_replica_greater_than_querynodes(self):
+        """
+        target: test load with replicas that greater than querynodes
+        method: load with 2 replicas (only 1 querynode)
+        expected: Raise exception
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        error = {ct.err_code: 1, ct.err_msg: f"no enough nodes to create replicas"}
+        collection_w.load(replica_number=2, check_task=CheckTasks.err_res, check_items=error)
+
+    @pytest.mark.xfail(reason="https://github.com/milvus-io/milvus/issues/16562")
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_load_replica_change(self):
+        """
+        target: test load replica change
+        method: 1.load with replica 1
+                2.load with a new replica number
+                3.release collection
+                4.load with a new replica
+        expected: The second time successfully loaded with a new replica number
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        collection_w.load(replica_number=1)
+        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
+        collection_w.load(replica_number=2)
```

Comment:
do we expect an error or warning here when loading with a different replicas

Reply:
I agree with throwing exceptions. However, there is a problem with the case itself. We can go ahead and modify it later